osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/datetime/SynchronizeTimeCommandExecutorTest.java

```diff
@@ -0,0 +1,114 @@
+
+@ExtendWith(MockitoExtension.class)
+class SynchronizeTimeCommandExecutorTest {
+
+  private static final ObisCode LOGICAL_NAME = new ObisCode("0.0.1.0.0.255");
+
+  private final DlmsDevice DLMS_DEVICE = new DlmsDevice();
+
+  @Captor ArgumentCaptor<SetParameter> setParameterArgumentCaptor;
+
+  @Mock private DlmsConnectionManager conn;
+
+  @Mock private DlmsMessageListener dlmsMessageListener;
+
+  @Mock private DlmsConnection dlmsConnection;
+
+  @Mock private MessageMetadata messageMetadata;
+
+  private SynchronizeTimeCommandExecutor executor;
+
+  @BeforeEach
+  void setUp() {
+    this.executor = new SynchronizeTimeCommandExecutor(new DlmsHelper());
+  }
+
+  @Test
+  void testSynchronizeTime() throws ProtocolAdapterException, IOException {
+    final String timeZone = "Europe/Amsterdam";
+    final ZonedDateTime expectedTime = ZonedDateTime.now(TimeZone.getTimeZone(timeZone).toZoneId());
+
+    // SETUP
+    when(this.conn.getDlmsMessageListener()).thenReturn(this.dlmsMessageListener);
+    when(this.conn.getConnection()).thenReturn(this.dlmsConnection);
+    when(this.dlmsConnection.set(any(SetParameter.class))).thenReturn(AccessResultCode.SUCCESS);
+
+    final SynchronizeTimeRequestDto synchronizeTimeRequest =
+        new SynchronizeTimeRequestDto(timeZone);
+
+    // CALL
+    final AccessResultCode resultCode =
+        this.executor.execute(
+            this.conn, this.DLMS_DEVICE, synchronizeTimeRequest, this.messageMetadata);
+
+    // VERIFY
+    assertThat(resultCode).isEqualTo(AccessResultCode.SUCCESS);
+    verify(this.dlmsConnection, times(1)).set(this.setParameterArgumentCaptor.capture());
+
+    final List<SetParameter> setParameters = this.setParameterArgumentCaptor.getAllValues();
+    assertThat(setParameters).hasSize(1);
+
+    final SetParameter setParameter = setParameters.get(0);
+    final AttributeAddress attributeAddress = setParameter.getAttributeAddress();
+    assertThat(attributeAddress.getClassId()).isEqualTo(InterfaceClass.CLOCK.id());
+    assertThat(attributeAddress.getInstanceId()).isEqualTo(LOGICAL_NAME);
+    assertThat(attributeAddress.getId()).isEqualTo(ClockAttribute.TIME.attributeId());
+
+    assertThat(setParameter.getData().getType().name()).isEqualTo("DATE_TIME");
+    final CosemDateTime cosemDateTime = setParameter.getData().getValue();
+    assertThat(cosemDateTime.get(Field.HOUR)).isEqualTo(expectedTime.getHour());
+    assertThat(cosemDateTime.get(Field.MINUTE))
+        .isBetween(expectedTime.getMinute(), expectedTime.getMinute() + 1);
+    assertThat(cosemDateTime.get(Field.SECOND))
+        .isBetween(expectedTime.getSecond(), expectedTime.getSecond() + 1);
+    assertThat(cosemDateTime.get(Field.DAY_OF_MONTH)).isEqualTo(expectedTime.getDayOfMonth());
+    assertThat(cosemDateTime.get(Field.MONTH)).isEqualTo(expectedTime.getMonthValue());
+    assertThat(cosemDateTime.get(Field.YEAR)).isEqualTo(expectedTime.getYear());
```

Comment:
Wouldn't it be nice to check the duration between expectedTime and cosemDateTime to be less than 1 second?
Now the test successed if time difference is as long as one whole minute, but fails if the one second difference is just crossing the hour.